documentation/src/main/asciidoc/user_guide/chapter-64-Hot_Rod_Protocol.adoc

```diff
@@ -1533,6 +1534,37 @@ converter. If custom marker is `2`, it represents the byte array, as returned
 by the converter.
 |==============================================================================
 
+===== Hot Rod Protocol 2.2
+
+.Infinispan versions
+TIP: This version of the protocol is implemented since Infinispan 7.2.2.Final.
+
+Added support for different time units.
+
+====== Operations
+
+.Put/PutAll/PutIfAbsent/Replace/ReplaceIfUnmodified
+
+Common request format:
+
+[cols="3,^2,10",options="header"]
+|==============================================================================
+| Field Name          | Size       | Value
+
+| Lifespan            | vLong       | Duration which the entry is allowed to life.
+| Max Idle            | vLong       | Duration that an entry can be
+idle before it's evicted from the cache. If 0, no max idle time.
+| TimeUnits           | Byte        | Time units of lifespan (first 4 bits) and maxIdle (last 4 bits). Possible values: +
++0x00+ = SECONDS +
++0x01+ = MILLISECONDS +
++0x02+ = NANOSECONDS +
++0x03+ = MICROSECONDS +
++0x04+ = MINUTES +
++0x05+ = HOURS +
++0x06+ = DAYS +
```

Comment:
I assume we'd need to get rid of the flags ```DEFAULT_LIFESPAN``` and ```DEFAULT_MAXIDLE``` since having a ```DEFAULT``` special unit would be enough?